server/schedulers/balance_leader.go

```diff
@@ -161,42 +164,92 @@ func (l *balanceLeaderScheduler) Schedule(cluster schedule.Cluster) []*operator.
 			targets[j].LeaderScore(leaderSchedulePolicy, jOp)
 	})
 
-	for i := 0; i < len(sources) || i < len(targets); i++ {
-		if i < len(sources) {
-			plan.source, plan.target = sources[i], nil
+	usedRegions := make(map[uint64]struct{})
+	sourcePoint := 0
+	targetPoint := 0
```

Comment:
I think at least some comments are needed.

Reply:
rafactor it